app/javascript/article-form/components/EditorBody.jsx

```diff
@@ -14,28 +14,86 @@ import { useDragAndDrop } from '@utilities/dragAndDrop';
 import { fetchSearch } from '@utilities/search';
 import { AutocompleteTriggerTextArea } from '@crayons/AutocompleteTriggerTextArea';
 
-function handleImageSuccess(textAreaRef) {
-  return function (response) {
+// Placeholder text displayed while an image is uploading
+const UPLOADING_IMAGE_PLACEHOLDER = '![Uploading image](...)';
```

Comment:
@rt4914 I tested this and it works pretty well. However I'm thinking since the contents of the `handleImageUpload`, `handleImageSuccess` and `handleImageUploadFailure` are duplicated across the `EditorBody` file and the `CommentTextArea` file that perhaps we should move it into a helper function and reference it in both those files instead. That way, in the future if we need to change the way that image uploading behaves, we only need to change it in a single file. 


What do you think?

Reply:
I agree with these suggestions, will fix this.